### ISSUE TITLE
Make requirements for plotting optional

### ISSUE THREAD

**Issue by hackermd** [keflavich/image_registration#16]
At the moment the library cannot be used without installing graphical libraries, such as Matplotlib or Tkinter.

For many non-interactive use cases, plotting functionality is not required (for example in a HPC environment) and should therefore be optional in my opinion.

**Comment by keflavich** [comment]
Fair enough, I think that can be achieved.  I'd welcome a PR to factor out the plotting utilities, otherwise I can try to get to it when I have time....

**Comment by davidedelvento** [comment]
Until that is done, the `requirements.txt` should be changed to have matplotlib included, rather than failing at runtime
